source/common/upstream/health_discovery_service.cc

```diff
@@ -54,12 +87,73 @@ void HdsDelegate::onReceiveInitialMetadata(Http::HeaderMapPtr&& metadata) {
   UNREFERENCED_PARAMETER(metadata);
 }
 
+void HdsDelegate::processMessage(
+    std::unique_ptr<envoy::service::discovery::v2::HealthCheckSpecifier>&& message) {
+  for (int i = 0; i < message->health_check_size(); i++) {
+    // Create HdsCluster config
+    ENVOY_LOG(debug, "Creating HdsCluster config");
+    envoy::api::v2::core::BindConfig bind_config;
+    envoy::api::v2::Cluster cluster_config;
+    clusters_config_.push_back(cluster_config);
+    clusters_config_[i].set_name(message->mutable_health_check(i)->cluster_name());
+    clusters_config_[i].mutable_connect_timeout()->set_seconds(cluster_timeout_s);
+    clusters_config_[i].mutable_per_connection_buffer_limit_bytes()->set_value(
+        cluster_connection_buffer_limit_bytes);
+
+    for (int j = 0; j < message->mutable_health_check(i)->endpoints_size(); j++) {
+      for (int k = 0; k < message->mutable_health_check(i)->mutable_endpoints(j)->endpoints_size();
```

Comment:
Instead of taking mutable references, you can take just the `const` ones here, as you are copying from. I.e.
`message.health_check(i).endpoints(j)` etc. Even better would be in the outer loop body to do:
`const auto& health_check = message.health_check(i);` and use that in the inner loop.